modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/mapping/QuerySplitter.java

```diff
@@ -188,12 +191,13 @@ public IgniteRel visit(IgniteTableScan rel) {
         IgniteTable table = rel.getTable().unwrap(IgniteTable.class);
 
         assert table != null;
+        long sourceId = idGenerator.nextId();
 
         if (curr.seenRelations.add(table.id())) {
-            curr.tables.add(table);
+            curr.tables.put(sourceId, table);
```

Comment:
Does this mean table will be added only once, the same table with different sourceId will be ignored and later can't be found?

Reply:
If so, then what is the purpose of using Long2ObjectMap?
Adding cost O(n), `get` costs also O(n). It is worst alternative to a hash-based map and a list.

Reply:
Thanks. fixed.

Reply:
BTW, you missed the very first question. Should `if` clause be removed?
As I understand, you need a full `sourceID-> table` mapping.
The same table may occurs with more than once with different `sourceId`, but you will never (I guess) see `sourceId` duplicates, isn't it?
```
 if (curr.seenRelations.add(table.id())) {
             curr.tables.put(sourceId, table);
```

Reply:
>you missed the very first question. Should if clause be removed?
No, it should not. The same table/view within query have the same id.